### PR TITLE
Update label edit

### PR DESCRIPTION
**Pull request by cazci**
# Description

Updated the label editing logic and added a check for label type update. If the user needs to update the label type, he has to manually review the selection instances of the particular label. Otherwise, he can simply update the label title and the selections will be updated accordingly. 

Related to #423 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Screenshots

### On Success

| Update dialog | Project details screen |
|---|---|
| Screenshot_1592933295 | Screenshot_1592933316 |

### On Error

| Update dialog | Project details screen |
|---|---|
| Screenshot_1592933183 | Screenshot_1592933213 |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by UdeshUK** [comment]
@cazci can't we cascade the updated label to the selections?

**Comment by cazci** [comment]
Sure, my guess for reviewing was to prevent deletion of any important selections automatically

**Comment by UdeshUK** [comment]
@cazci In the case of editing label type, I think we should not allow it. Because even if allow it the user will lose all the selections done with previous type and that no different than deleting the label and creating it.

**Comment by cazci** [comment]
Yes true, I'll update it accordingly

File: labellab_mobile/lib/screen/project/detail/project_detail_screen.dart [hunk 1]
```diff
@@ -560,15 +564,15 @@ class ProjectDetailScreen extends StatelessWidget {
...
+              Provider.of<ProjectDetailBloc>(baseContext).deleteLabel(label.id);
               Navigator.of(context).pop(true);
             },
           );
         });
   }
 
-  void _showAddEditLabelModel(BuildContext baseContext, Label label) {
+  void _showAddEditLabelModel(BuildContext baseContext,
+      {String projectId, List<LabelLab.Image> images, Label label}) {
```
Comment (UdeshUK) [inline labellab_mobile/lib/screen/project/detail/project_detail_screen.dart:575]: @cazci Image argument is unused here. Is there a reason for this?

Comment (ivantha) [reply labellab_mobile/lib/screen/project/detail/project_detail_screen.dart:575]: @cazci Can you address this, please?

**Comment by UdeshUK** [comment]
@cazci please rebase to resolve confilcts

**Pull request merged**